server/src/main/java/com/google/moviestvsentiments/webscrape/AssetScraper.java

```diff
@@ -0,0 +1,152 @@
+        WebClient client = new WebClient();
+        client.getOptions().setCssEnabled(false);
+        client.getOptions().setJavaScriptEnabled(false);
+        HtmlPage page = client.getPage(url);
+        List<HtmlAnchor> assetLinks = page.getByXPath(IMDB_ASSET_XPATH);
+
+        List<String> assetIds = new ArrayList<>();
+        for (HtmlAnchor assetLink : assetLinks) {
+            String assetUrl = assetLink.getHrefAttribute();
+            int startPosition = IMDB_ASSET_URL_PREFIX.length();
+            int endPosition = assetUrl.lastIndexOf("/");
+            assetIds.add(assetUrl.substring(startPosition, endPosition));
+        }
+        return assetIds;
+    }
+
+    /**
+     * Returns a list of Assets scraped from OMDB. The given OMDB api key and the list of Asset ids is used to query
+     * OMDB for the Assets.
+     * @param assetIds The list of Asset ids to query OMDB for.
+     * @param omdbApiKey The api key to use when querying OMDB.
+     * @return A list of Assets scraped from OMDB.
+     * @throws IOException If the OMDB query fails.
+     */
+    public List<Asset> scrapeAssets(List<String> assetIds, String omdbApiKey) throws IOException {
+        List<Asset> assets = new ArrayList<>();
+        for (String assetId : assetIds) {
+            String url = "http://www.omdbapi.com/?plot=full&apikey=" + omdbApiKey + "&i=" + assetId;
+            HttpRequest request = requestFactory.buildGetRequest(new GenericUrl(url));
+            OmdbResponse response = request.execute().parseAs(OmdbResponse.class);
+            assets.add(createAsset(response));
+        }
+        return assets;
+    }
+
+    /**
+     * Creates a new Asset from the given OmdbResponse. The Asset's timestamp is set to the current time.
+     * @param omdbResponse The OmdbResponse to use when creating the Asset.
+     * @return A new Asset created using the OmdbResponse.
+     */
+    private Asset createAsset(OmdbResponse omdbResponse) {
+        Asset asset = new Asset();
+        asset.setAssetId(omdbResponse.assetId);
+        asset.setTitle(omdbResponse.title);
+        asset.setPoster(omdbResponse.poster);
+        asset.setImdbRating(omdbResponse.imdbRating);
+        asset.setPlot(omdbResponse.plot);
+        asset.setRuntime(omdbResponse.runtime);
+        asset.setYear(omdbResponse.year);
+        asset.setTimestamp(Instant.now(clock));
+
+        if (OMDB_MOVIE_TYPE.equals(omdbResponse.assetType)) {
+            asset.setAssetType(AssetType.MOVIE);
+        } else if (OMDB_SHOW_TYPE.equals(omdbResponse.assetType)) {
+            asset.setAssetType(AssetType.SHOW);
+        } else {
+            throw new AssertionError("Unknown OMDB asset type: " + omdbResponse.assetType);
+        }
+
+        for (OmdbResponse.Rating rating : omdbResponse.ratings) {
+            if (OMDB_ROTTEN_TOMATOES_SOURCE.equals(rating.source)) {
+                asset.setRottenTomatoesRating(rating.value);
+            }
+        }
```

Comment:
I think this can be written with stream

omdbResponse.ratings.stream().filter(rating -> OMDB_ROTTEN_TOMATOES_SOURCE.equals(rating.source)).findAny().ifPresent(rating -> asset.setRottenTomatoesRating(rating.value));